deployment/base/hg-deployment.yaml

```diff
@@ -131,3 +131,33 @@ spec:
           items:
           - key: hgweb.hgrc
             path: hgweb.hgrc
+
+      initContainers:
+      - name: init-repo-structure
+        securityContext:
+          runAsUser: 33
+          runAsGroup: 33 # www-data
+        image: busybox:1.36.1
+        command:
+          - 'sh'
+          - '-c'
+          - |
+            cd /repos
+            for folder in /repos/[a-z0-9]?*/; do
+              project=$(basename "$folder")
+              if [[ -L "$project" ]]; then
+                echo "Skipping symlinked folder: $project"
+                continue
+              fi
+              first_letter=$(echo "$project" | head -c1)
```

Comment:
`cut -c1` is the usual idiom rather than `head -c1`, but both will work as long as you're dealing with single-line input. In fact, `head -c1` can theoretically be better if you have filenames with newlines in the middle of them (as then `cut -c1` would grab the first letter of "both" lines), but that's not going to happen to our repos so it's nearly a moot point.